fhir-search/src/main/java/com/ibm/fhir/search/reference/value/TypeIdHandlerImpl.java

```diff
@@ -6,18 +6,26 @@
 package com.ibm.fhir.search.reference.value;
 
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import com.ibm.fhir.search.parameters.QueryParameterValue;
 
 /**
  * The supported cases are:
  * Patient/1
  * -> http://server/Patient/1
+ * -> 1 (only when there is one target)
  */
 public class TypeIdHandlerImpl implements ParameterValueHandler {
+
+    private static final String REGEX = "^([A-z][a-z]{2,36}/[A-Za-z0-9\\-\\.]{1,64})$";
```

Comment:
same comment about erring on the side of caution on referencetype length
```suggestion
    private static final String REGEX = "^([A-z][a-z]{2,64}/[A-Za-z0-9\\-\\.]{1,64})$";
```